CMakeLists.txt

```diff
@@ -88,15 +103,17 @@ add_subdirectory(src)
 #
 add_subdirectory(srcpy)
 
-#
-# manage the demos
-#
-add_subdirectory(demos)
+if (BUILD_ROS_DYNAMIC_GRAPH)
+    #
+    # manage the demos
+    #
+    add_subdirectory(demos)
+endif()
 
 #
 # manage the unit tests
 #
-if(BUILD_TESTING)
+if (BUILD_ROS_DYNAMIC_GRAPH AND BUILD_TESTING)
```

Comment:
the if  should be inside the `tests/CMakeLists.txt` here

Reply:
I addressed this one.